### DISCUSSION TITLE
Error running migrations

### DISCUSSION THREAD

**Discussion by JensSpanier** [dani-garcia/vaultwarden#3676] [Q&A] [answered]
### Subject of the issue
When I try to start the new version `1.29.0`, I get this error:
```
[2023-07-09 17:39:11.931][panic][ERROR] thread 'main' panicked at 'Error running migrations: QueryError(DieselMigrationName { name: "2023-06-02-200424_create_organization_api_key", version: MigrationVersion("20230602200424") }, DatabaseError(Unknown, "Can't create table `jspanier_vaultwarden`.`organization_api_key` (errno: 150 \"Foreign key constraint is incorrectly formed\")"))': src/db/mod.rs:471
```

### Deployment environment

* vaultwarden version: `1.29.0`

* Install method: Docker image

* MariaDB version: `10.6.12`

* Other relevant details:

### Troubleshooting data
I previously updated an instance successfully which has no organisations (MariaDB-Version `10.3.38`).
This instance has two organisations.

Let me know if you need more information.

**Comment by BlackDex** [comment, marked as answer]
See https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend#foreign-key-errors-collation-and-charset

**Reply by JensSpanier** [reply]
Thank you! What's the correct collation? `utf8mb4_unicode_ci` or `utf8mb4_general_ci`?

The wiki says
> It should return `utf8mb4` and `utf8mb4_general_ci`.

But all commands include `utf8mb4_unicode_ci`.

**Reply by BlackDex** [reply]
It should be `utf8mb4_unicode_ci`, thanks for noticing, i fixed the wiki.

**Reply by JensSpanier** [reply]
```SQL
ALTER DATABASE `vaultwarden` CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
fixed the problem. The tables were already in the correct collation.

Thank you for your quick support!
